cpp/include/cudf/io/types.hpp

```diff
@@ -342,8 +342,8 @@ struct source_info {
    *
    * @param file_paths Input files paths
    */
-  explicit source_info(std::vector<std::string> const& file_paths)
-    : _type(io_type::FILEPATH), _filepaths(file_paths)
+  explicit source_info(std::vector<std::string> file_paths)
+    : _type(io_type::FILEPATH), _filepaths(std::move(file_paths))
```

Comment:
We use this idiom (pass by value into std::move) extensively in the options classes. I think this code predates the options so it makes unnecessary copies. Thanks for the update!